### PR TITLE
VideoPress: check if it's an auto-generated chapters file

### LINKED ISSUE #26263: VideoPress: add video-chapter to v6

### LINKED ISSUE #27453: VideoPress: check chapters file before to overwrites it
> a vtt file should always win, and we should render a notice below the description if we detect chapters in the description saying “you already have a vtt file”

### PR DESCRIPTION
**Pull request by retrofox**
This PR checks if the video has a not auto-generated chapter .vtt file no avoid overwriting it.

Fixes https://github.com/Automattic/jetpack/issues/26263

#### Changes proposed in this Pull Request:

* VideoPress: check if chapters .vtt file exists

#### Testing instructions:

* Go to Block editor
* Since uploading a vtt file is not migrated to v6 yet, use a `core/video` extended video block (Jetpack plugin) to upload the file
image

image

image

* Save the post
* Hard refresh
* Confirm the video has chapters
Screen Shot 2022-11-22 at 07 56 50

* Transform the block to v6
* Open block sidebar
* Confirm it shows a notice about the vtt chapters file

Screen Shot 2022-11-22 at 07 59 32

* Confirm even when the description has chapters, it does not overwrite the current one.

* Create a new block with a video (ensure it doesn't have chapters)
* Confirm you do not see the notice
* Confirm you can add chapters

### CODE REVIEW

**Comment by lhkowalski** [comment]
I followed the testing instructions. After I added a VTT file as chapters on the video, I could see the notice on the right panel:

Screen Shot 2022-11-22 at 19 09 16

I changed the video description to include some chapters, but they do not reflected on the player (as expected):

Screen Shot 2022-11-22 at 19 11 04

I added another block using a video without chapters, and the notice is not there (as expected):

Screen Shot 2022-11-22 at 19 20 05

I changed the video description for this other block, including some chapters, and they got applied to the player (as expected):

Screen Shot 2022-11-22 at 19 21 05

So it's working as described on the testing instructions. I will now try some edge cases to see if I can break things :)

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.tsx [hunk 1]
```diff
@@ -72,6 +75,14 @@ export default function DetailsPanel( {
 				help={ descriptionHelp }
 			/>
 
+			{ !! chapter && ! isAutoGeneratedChapter && (
+				<Notice status="success" className="learn-how-notice" isDismissible={ false }>
+					<p className="learn-how-notice__message">
+						{ __( 'You already have a vtt file', 'jetpack-videopress-pkg' ) }
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.tsx]: Should we be more informative here, like "You already have chapter information on a attached VTT file. Adding chapters to the description will not change the original chapters", so the user knows that the chapters won't work from there?

File: projects/packages/videopress/src/client/utils/generate-chapters-file/index.ts [hunk 2]
```diff
@@ -37,6 +39,15 @@ function generateChaptersFileContent(
 	}
 
 	let content = 'WEBVTT\n';
+
+	// Add a tag to identify auto-generated chapters
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/utils/generate-chapters-file/index.ts:43]: Could we prevent doing this by setting some metadata on the video to say that the chapters were autogenerated? I like this file-change approach (the info is embed on the file itself), just want to make sure we explored other options :)

Comment (retrofox) [reply projects/packages/videopress/src/client/utils/generate-chapters-file/index.ts:43]: Thanks, Luiz. We wanted to tackle this by adding a prefix to the vtt filename when it's auto-generated, but the API doesn't support setting it.
Using metadata sounds good to me. Reading the file seems more confident, but I don't believe in the best approach.
Maybe we can try it in a follow-up task.

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.tsx (hunk 1 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.tsx]: we've updated the description [after talking with VP folks](p1669201017490559-slack-C02LT75D3).

image

Addressed [here](https://github.com/Automattic/jetpack/pull/27544/commits/eeab29ee7bd8a88c80092cb392517b68bda8da30)

File: projects/packages/videopress/src/client/utils/generate-chapters-file/index.ts (hunk 2 above)
Comment (lhkowalski) [reply projects/packages/videopress/src/client/utils/generate-chapters-file/index.ts:43]: I agree, and I actually like the actual approach, it's self contained.

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.tsx [hunk 3]
```diff
@@ -72,6 +82,24 @@ export default function DetailsPanel( {
 				help={ descriptionHelp }
 			/>
 
+			{ !! chapter && ! isAutoGeneratedChapter && (
+				<Notice status="success" className="learn-how-notice" isDismissible={ false }>
+					<p className="learn-how-notice__message">
+						{ createInterpolateElement(
+							__(
+								'You already have chapter information on a attached VTT file, so adding chapters to the description will not change the original ones. <link>Learn more</link>',
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.tsx]:
```suggestion
								'You already have chapter information on an attached VTT file, so adding chapters to the description will not change the original ones. <link>Learn more</link>',
```

**Review by lhkowalski — DISMISSED** [review]
Great work! 🚀

**Review by lhkowalski — APPROVED** [review]
Good catch on the tests!

**Pull request merged**